setup-runner-linux.sh

```diff
@@ -71,19 +67,21 @@ echo "================= Configuring SSH. ================="
 
 # Adding "Subsystem powershell /usr/bin/pwsh -sshs -NoLogo -NoProfile" to /etc/ssh/sshd_config if its not there
 if grep -q "Subsystem powershell /usr/bin/pwsh -sshs -NoLogo -NoProfile" /etc/ssh/sshd_config; then
-  echo "================= Subsystem powershell is already present in /etc/ssh/sshd_config. ================="
+  echo "================= Subsystem powershell is already present in /etc/ssh/sshd_config. Skipping configuring SSH. ================="
 else
+# NOTE: Need to further investigate permissions and ideally remove the 777 chmod.
   echo "================= Adding 'Subsystem powershell /usr/bin/pwsh -sshs -NoLogo -NoProfile' to /etc/ssh/sshd_config. ================="
   echo "Subsystem powershell /usr/bin/pwsh -sshs -NoLogo -NoProfile" | sudo tee -a /etc/ssh/sshd_config
   echo ">>> installing sshpass"
   sudo apt install sshpass -y
   echo ">>> Adding new SSH key pair to netperf peer"
-  ssh-keygen -t rsa -b 2048 -N "" -f $HOME/.ssh/id_rsa
-  sshpass -p $password ssh-copy-id -o StrictHostKeyChecking=no $username@$peerip
+  ssh-keygen -t rsa -N "" -f $HOME/.ssh/id_rsa
+  sudo sshpass -p $password ssh-copy-id -i $HOME/.ssh/id_rsa.pub -o StrictHostKeyChecking=no $username@$peerip
+  chmod 777 $HOME/.ssh/id_rsa
   sudo systemctl restart ssh
 fi
 
-# Installing powershell 7
+# Installing powershell 7 (NOTE: Sometimes, will have to run this again because powershell fails to install the first run. Need investigation.)
```

Comment:
Put it in a loop, and keep running if `pwsh` doesn't work?

Reply:
Great idea, I wonder if I have to refresh my shell or bash does something weird... A follow PR is in the books.